### DISCUSSION TITLE
[React Native Expo - Migrate from react-native-dapp using modal-react-native] Failed to construct 'TextDecoder': the 'fatal' option is unsupported.

### DISCUSSION THREAD

**Discussion by jupiterrosen** [WalletConnect/walletconnect-monorepo#3100] [Developer Support]
![image](https://github.com/WalletConnect/walletconnect-monorepo/assets/112633538/4dca3314-22c0-4e2f-9a15-3a12bbf32d78)
This error appears once I have imported "useWalletConnectModal"

"dependencies": {
    "@react-native-async-storage/async-storage": "^1.19.0",
    "@solana/web3.js": "1.41.6",
    "@walletconnect/encoding": "^1.0.2",
    "@walletconnect/modal-react-native": "^1.0.0-rc.6",
    "big-integer": "^1.6.51",
    "bs58": "^5.0.0",
    "expo": "^46.0.17"
    "react-native": "0.69.6",
}

**Comment by ignaciosantise** [comment]
hey @jupiterrosen can you provide a repository with the reproducible issue?

**Reply by jupiterrosen** [reply]
It's ok now I have solved it

**Reply by ignaciosantise** [reply]
@jupiterrosen great! can you add your solution here? For future readers 🤝

**Reply by jupiterrosen** [reply]
Upgrade expo to the latest version

**Comment by mwaleed-39** [comment]
I faced some issue in react native. when we pressed upload image, the error message shows:

How to call this API, when we upload image in front end side, the image will be stored in mongodb
I create API:
[app.post](http://app.post/)('/finerequest', (req, res, next) => {

const file = req.files.image;
cloudinary.uploader.upload(file.tempFilePath, (err, result) => {
console.log(result)
const requestId = req.body.request_id
const fine = new Fine({
_id: new mongoose.Types.ObjectId,
request_id: requestId,
image: result.url
})
fine.save()
.then(result => {
res.status(200).json({
message: "Fine and Vehicle Repair Request added",
response: result
})
})
.catch(err => {
res.status(500).json({
message: "Fine and Vehicle request not added",
error: err
})
})
})
})

CODE:

import React, { useState } from 'react'
import { Alert, Button, View,StyleSheet } from 'react-native'
import * as ImagePicker from 'react-native-image-picker'
import {launchImageLibrary} from 'react-native-image-picker'

const options ={
title: 'Select Image',
type: 'library',
options: {
maxHeight:200,
maxWidth:200,
selectionLimit: 1,
mediaType: 'photo',
includeBase64: false,
}
}

const Request = () => {
const opengallery = async()=>{
const image = await ImagePicker.launchImageLibrary(options);
console.log(image)
}

return (

)
}
export default Request;

const styles = StyleSheet.create({
container:{
flex:1,
alignItems:'center',
justifyContent:'center'
}
})

when we pressed upload button, the error message shows:

Possible Unhandled Promise Rejection (id: 1):
TypeError: Cannot read property 'launchImageLibrary' of null
TypeError: Cannot read property 'launchImageLibrary' of null

Again and again same error show, we Image picker library, I think it has been deprecated. Provide Solution


Image pIcker or launchImageLibrary are not working
